File: backend/src/models/Document.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, DeleteDateColumn } from 'typeorm';

export enum DocumentStatus {
    UPLOADING = 'uploading',
    PROCESSING = 'processing',
    COMPLETED = 'completed',
    FAILED = 'failed'
}

export enum FileType {
    PDF = 'pdf',
    DOCX = 'docx',
    TXT = 'txt',
    MD = 'md',
    IMAGE = 'image',
    VIDEO = 'video',
    AUDIO = 'audio',
    OTHER = 'other'
}

@Entity('documents')
export class Document {
    @PrimaryGeneratedColumn('uuid')
    id!: string;

    @Column({ name: 'user_id', type: 'uuid' })
    userId!: string;

    @Column({ name: 'workspace_id', type: 'uuid', nullable: true })
    workspaceId?: string;

    // Content
    @Column({ length: 500 })
    title!: string;

    @Column({ type: 'text', nullable: true })
    content?: string;

    @Column({ type: 'text', nullable: true })
    summary?: string;

    // File metadata
    @Column({ name: 'file_path', length: 1000, nullable: true })
    filePath?: string;

    @Column({ name: 'file_name', length: 255, nullable: true })
    fileName?: string;

    @Column({ name: 'file_size_bytes', type: 'bigint', nullable: true })
    fileSizeBytes?: number;

    @Column({ name: 'file_type', type: 'enum', enum: FileType, nullable: true })
    fileType?: FileType;

    @Column({ name: 'mime_type', length: 100, nullable: true })
    mimeType?: string;

    // Processing status
    @Column({
        type: 'enum',
        enum: DocumentStatus,
        default: DocumentStatus.UPLOADING
    })
    status!: DocumentStatus;

    @Column({ name: 'processing_error', type: 'text', nullable: true })
    processingError?: string;

    // Timestamps
    @CreateDateColumn({ name: 'created_at' })
    createdAt!: Date;

    @UpdateDateColumn({ name: 'updated_at' })
    updatedAt!: Date;

    @DeleteDateColumn({ name: 'deleted_at', nullable: true })
    deletedAt?: Date;
}

File: ai-knowledge-system/src-tauri/src/services/document.rs
use crate::models::{Document, CreateDocumentDto, DocumentStatus};
use sqlx::PgPool;
use uuid::Uuid;

pub struct DocumentService {
    pool: PgPool,
}

impl DocumentService {
    pub fn new(pool: PgPool) -> Self {
        DocumentService { pool }
    }
    
    pub async fn create_document(&self, dto: CreateDocumentDto) -> Result<Document, sqlx::Error> {
        let doc = sqlx::query_as!(
            Document,
            r#"
            INSERT INTO documents (
                user_id, title, file_name, file_size_bytes, file_type, mime_type, status
            )
            VALUES ($1, $2, $3, $4, $5, $6, 'uploading')
            RETURNING 
                id, user_id, workspace_id, title, content, summary, 
                file_path, file_name, file_size_bytes, file_type, mime_type,
                status as "status!: DocumentStatus",
                processing_error, created_at, updated_at, deleted_at
            "#,
            dto.user_id,
            dto.title,
            dto.file_name,
            dto.file_size_bytes,
            dto.file_type,
            dto.mime_type
        )
        .fetch_one(&self.pool)
        .await?;
        
        Ok(doc)
    }
    
    pub async fn update_file_path(&self, doc_id: Uuid, file_path: String) -> Result<(), sqlx::Error> {
        sqlx::query!(
            r#"
            UPDATE documents
            SET file_path = $2, updated_at = NOW()
            WHERE id = $1
            "#,
            doc_id,
            file_path
        )
        .execute(&self.pool)
        .await?;
        
        Ok(())
    }
    
    pub async fn update_content_and_summary(
        &self,
        doc_id: Uuid,
        content: String,
        summary: String,
    ) -> Result<(), sqlx::Error> {
        sqlx::query!(
            r#"
            UPDATE documents
            SET content = $2, summary = $3, status = 'completed', updated_at = NOW()
            WHERE id = $1
            "#,
            doc_id,
            content,
            summary
        )
        .execute(&self.pool)
        .await?;
        
        Ok(())
    }
    
    pub async fn get_documents_by_user(&self, user_id: Uuid) -> Result<Vec<Document>, sqlx::Error> {
        let docs = sqlx::query_as!(
            Document,
            r#"
            SELECT 
                id, user_id, workspace_id, title, content, summary,
                file_path, file_name, file_size_bytes, file_type, mime_type,
                status as "status!: DocumentStatus",
                processing_error, created_at, updated_at, deleted_at
            FROM documents
            WHERE user_id = $1 AND deleted_at IS NULL
            ORDER BY created_at DESC
            "#,
            user_id
        )
        .fetch_all(&self.pool)
        .await?;
        
        Ok(docs)
    }
    
    pub async fn update_document_status(
        &self,
        doc_id: Uuid,
        status: DocumentStatus,
        error: Option<String>,
    ) -> Result<(), sqlx::Error> {
        sqlx::query!(
            r#"
            UPDATE documents
            SET status = $2, processing_error = $3, updated_at = NOW()
            WHERE id = $1
            "#,
            doc_id,
            status as DocumentStatus,
            error
        )
        .execute(&self.pool)
        .await?;
        
        Ok(())
    }
}
